src/types/string.rs

```diff
@@ -172,6 +185,20 @@ impl<'source> FromPyObject<'source> for String {
     }
 }
 
+impl<'source> FromPyObject<'source> for char {
+    fn extract(obj: &'source PyAny) -> PyResult<Self> {
+        let s = PyString::try_from(obj)?.to_str()?;
+        let mut iter = s.chars();
+        if let (Some(ch), None) = (iter.next(), iter.next()) {
+            Ok(ch)
+        } else {
+            Err(crate::exceptions::PyValueError::new_err(format!(
+                "Expected a sting of length 1",
```

Comment:
typo `sting` -> `string`, and I think the recommendation in https://github.com/PyO3/pyo3/pull/1212#issuecomment-707557245 was that it's better for us to make errors start with lowercase
```suggestion
                "expected a string of length 1",
```